src/main/java/org/folio/services/validator/registry/ValidatorRegistryServiceImpl.java

```diff
@@ -0,0 +1,216 @@
+package org.folio.services.validator.registry;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.rest.jaxrs.model.Rule;
+import org.folio.rest.jaxrs.model.RuleCollection;
+import org.folio.rest.persist.Criteria.Criteria;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.persist.cql.CQLWrapper;
+import org.z3950.zing.cql.cql2pgjson.CQL2PgJSON;
+
+import java.util.List;
+import java.util.UUID;
+
+public class ValidatorRegistryServiceImpl implements ValidatorRegistryService {
+
+  private final Logger logger = LoggerFactory.getLogger(ValidatorRegistryServiceImpl.class);
+
+  private static final String VALIDATION_RULES_TABLE_NAME = "validation_rules";
+  private static final String RULE_ID_FIELD = "ruleId";
+  private static final String RULE_ID_JSONB_FIELD = "'ruleId'";
+  private static final String STATE_JSONB_FIELD = "'state'";
+  private static final String TYPE_JSONB_FIELD = "'type'";
+
+  private final Vertx vertx;
+
+  public ValidatorRegistryServiceImpl(Vertx vertx) {
+    this.vertx = vertx;
+  }
+
+  /**
+   * Returns all rules for tenant
+   *
+   * @param tenantId           tenant id
+   * @param asyncResultHandler result handler
+   * @return a reference to this, so the API can be used fluently
+   */
+  @Override
+  public ValidatorRegistryService getAllTenantRules(String tenantId, Handler<AsyncResult<JsonObject>> asyncResultHandler) {
+    try {
+      CQL2PgJSON cql2pgJson = new CQL2PgJSON(VALIDATION_RULES_TABLE_NAME + ".jsonb");
+      CQLWrapper cql = new CQLWrapper();
+      cql.setField(cql2pgJson);
+      String[] fieldList = {"*"};
+      PostgresClient.getInstance(vertx, tenantId).get(VALIDATION_RULES_TABLE_NAME, Rule.class, fieldList, cql, true, false, getReply -> {
+        if (getReply.failed()) {
+          logger.error("Error while querying the db to get all tenant rules", getReply.cause());
+          asyncResultHandler.handle(Future.failedFuture(getReply.cause()));
+        } else {
+          RuleCollection rules = new RuleCollection();
+          List<Rule> ruleList = (List<Rule>) getReply.result().getResults();
+          rules.setRules(ruleList);
+          rules.setTotalRecords(ruleList.size());
+          asyncResultHandler.handle(Future.succeededFuture(JsonObject.mapFrom(rules)));
+        }
+      });
+    } catch (Exception e) {
+      logger.error("Error while getting all tenant rules", e.getCause());
```

Comment:
e.getCause() may be null if thrown from PostgresClient.getInstance or PostgresClient.getInstance(vertx, tenantId).get

Reply:
@julianladisch, thank you for pointing this out